lbry/dht/protocol/iterative_find.py

```diff
@@ -182,13 +190,17 @@ def _search_round(self):
         added = 0
         for index, peer in enumerate(self.active.keys()):
             if index == 0:
-                log.debug("closest to probe: %s", peer.node_id.hex()[:8])
+                log.debug("%s[%x] closest to probe: %s",
+                          type(self).__name__, id(self),
+                          peer.node_id.hex()[:8])
             if peer in self.contacted:
                 continue
             if len(self.running_probes) >= constants.ALPHA:
                 break
             if index > (constants.K + len(self.running_probes)):
                 break
+            if self.iteration_count + self.iteration_queue.qsize() >= self.max_results:
```

Comment:
Is this related to the current bug? If not, it would be cool to separate in another PR

Reply:
When I started looking at this, it seemed questionable that max_results was not considered before doing more work. These might be extra results which would not be consumed. But I see now the recent changes to iterative_find.py in Mar and Dec which removed max_results from consideration in some cases. And it's unclear when/where a caller might utilize max_results=N -- existing code defaults to K or K*2.

For this particular key (2ecce30223bc2808bd10...) it seems like search-exhausted comes from this bit of code even with large max_results.

`
         if index > (constants.K + len(self.running_probes)):
             break
`

Small max_results (<K) is ignored too (`self.max_results = max(constants.K, max_results)`)

Since I have little understanding of the algorithm and purpose of max_results, I will drop code limiting output.